### PR TITLE
PCHR-3166: Make deduction selectors look like dropdowns

### PR DESCRIPTION
**Pull request by igorpavlov**
## Overview

This PR makes Deduction fields in the Leave Request Modal look more like selectors.

## Before

image

### Admin

image

### SSP

image

## Technical Details

```html
<div class="time-amount-picker">
  <div class="crm_custom-select"> <!-- wrap -->
    <select> ... </select>
    <span class="crm_custom-select__arrow"></span> <!-- add arrow -->
  </div>
  h
  <!-- same for mm -->
</div>
```

```css
.time-amount-picker {

  .crm_custom-select {
    display: inline-block; // Selectors need to appear in a single line
    margin-right: 2px; // A gap between a dropdown and a label

    &:nth-child(2) {
      margin-left: 10px; // A gap between dropdowns
    }
  }
}
```

## Comments

BackstopJS scenario is added:
image

*Please ignore the broken menu - this is a known bug*.

----

✅ Manual Tests - passed (SSP, Admin)
✅ JS distributive files - included
✅ CSS distributive files - included
✅ Jamie Test - passed
✅ Backstop Tests - amended and passed
⏹ PHP Unit Tests - not needed
⏹ Jasmine Tests - not needed

### CODE REVIEW

**Comment by jamienovick** [comment]
✅ Jamie test - included and passed; but we will be tweaking this anyway it seems! :)

File: uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js [hunk 1]
```diff
@@ -24,8 +25,33 @@ module.exports = (function () {
...
+     *
+     * @param  {String} requestType leave|sickness|toil
+     * @return {Promise}
+     */
+    newRequest: function (requestType) {
+      var casper = this.casper;
+      var requestTypes = ['leave', 'sickness', 'toil']; // must be in the same quantity and order as in UI
+      var requestTypeButtonIndex = requestTypes.indexOf(requestType) + 1;
+      var actionDropdownSelector = 'leave-request-record-actions';
+      var actionButtonSelector = actionDropdownSelector + ' .dropdown-menu a:nth-child(' + requestTypeButtonIndex + ')';
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js]: If you use jQuery, you can use the `.contains` method. For example you could do something like this:

```js
casper.evaluate(function (actionDropdownSelector, requestType) {
  var requestTypeButton = jQuery(actionDropdownSelector + ' .dropdown-menu a:contains(' + requestType + ')');

  requestTypeButton.click(); 
}, actionDropdownSelector, requestType);
```

File: uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js [hunk 2]
```diff
@@ -57,8 +84,38 @@ module.exports = (function () {
...
+     * @param  {String} absenceTypeLabel ex. "Holiday in Hours"
+     * @return {Promise}
+     */
+    selectRequestAbsenceType: function (absenceTypeLabel) {
+      var absenceTypeSelect, absenceTypeOptionIndex;
+      var casper = this.casper;
+
+      casper.then(function () {
+        casper.evaluate(function(absenceTypeLabel) {
+          absenceTypeSelect = document.querySelector('[name=absenceTypeSelect]');
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js]: you could use jQuery here too:

```js
casper.evaluate(function (absenceTypeLabel) {
  var absenceTypeSelect = jQuery('[name=absenceTypeSelect]');
  var absenceTypeOption = absenceTypeSelect.find('option:contains(' + absenceTypeLabel + ')');

  // .change() triggers the change event for the select:
  absenceTypeSelect.val(absenceTypeOption.attr('value')).change();
, absenceTypeLabel);
```

File: uk.co.compucorp.civicrm.hrcore/backstop_data/casper_scripts/ssp-leave-absences/staff/my-report-show-absence-in-hours.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+var page = require('../../../page-objects/ssp-leave-absences-my-leave-report');
+
+// precondition: need to have the absence type in *hours* with a label "Holiday in Hours"
```
[the remark this answers is not in the document]
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrcore/backstop_data/casper_scripts/ssp-leave-absences/staff/my-report-show-absence-in-hours.js]: Yes, already noted to myself in the ticket. I cannot document this until it is on Staging.

File: uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js (hunk 1 above)
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js]: The whole point was not to use jQuery.

File: uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js (hunk 2 above)
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js]: I am not sure we want to have that dependency. I know jQuery is everywhere, but I prefer native code in this case.

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/directives/_time-amount-picker.scss [hunk 4]
```diff
@@ -1,4 +1,12 @@
-.time-amount-picker select {
-   padding-right: 12px;
-   width: 44px;
+.time-amount-picker {
+
+  .crm_custom-select {
+    display: inline-block; // Selectors need to appear in a single line
+    margin-right: 2px; // A gap between a dropdown and a label
+    width: 75px; // This width is comfortable with 2 digits inside
```
[the remark this answers is not in the document]
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrleaveandabsences/scss/directives/_time-amount-picker.scss]: `em` is related to **font-size**, no? maybe you meant `ch`?

File: uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js (hunk 2 above)
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js]: Rewrote as @AkA84 suggested

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/directives/_time-amount-picker.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrleaveandabsences/scss/directives/_time-amount-picker.scss]: I removed width completely.

**Pull request merged**